packages/astro/src/cli/check/index.ts

```diff
@@ -1,101 +1,360 @@
 /* eslint-disable no-console */
-import { AstroCheck, DiagnosticSeverity } from '@astrojs/language-server';
+import { AstroCheck, DiagnosticSeverity, GetDiagnosticsResult } from '@astrojs/language-server';
 import type { AstroSettings } from '../../@types/astro';
 import type { LogOptions } from '../../core/logger/core.js';
-
 import glob from 'fast-glob';
 import * as fs from 'fs';
 import { bold, dim, red, yellow } from 'kleur/colors';
 import { createRequire } from 'module';
 import ora from 'ora';
 import { fileURLToPath, pathToFileURL } from 'url';
 import { printDiagnostic } from './print.js';
+import type { Arguments as Flags } from 'yargs-parser';
+import { debug, info } from '../../core/logger/core.js';
+import type { SyncOptions, ProcessExit } from '../../core/sync';
+import fsMod from 'fs';
+import type { FSWatcher } from 'chokidar';
+import { join } from 'node:path';
 
-interface Result {
+type DiagnosticResult = {
```

Comment:
Any reasons for using a type instead of an interface here?

Reply:
`interface` comes with the feature to use `extends`/`implements`, which is not a feature I'm interested in. Generally, `type` is "lighter" compared to interfaces. Usually, I use `type` unless I need the interface features!